workspace-grid@mathematical.coffee.gmail.com/extension.js

```diff
@@ -323,11 +350,25 @@ function moveWorkspace(direction) {
  * This is the same as WindowManager._showWorkspaceSwitcher but we don't
  * filter out RIGHT/LEFT actions like they do.
  */
-function showWorkspaceSwitcher(display, screen, window, binding) {
+function showWorkspaceSwitcher(display, arg2, arg3, arg4) {
+    let screen;
+    let window;
+    let binding;
+    // Note: in v3.30, the 2nd arg (screen that we don't need) has been removed
+    if (!arg4) {
```

Comment:
isn't there any way to query the current gnome version rather than relying on a method signature?

Reply:
@zakkak I guess we can do that. I admit that's the first time I am modifying a GNOME Shell Extension and I don't know what's the common policy to support new versions. Is the goal to keep the code simple by not adding new 'if' conditions and then only support latest version? Or by looking at the signatures / available variables? Or by looking at the version?

If I need to look at the version, how can I do that? (I am not even sure where to find the official doc :) )
My changes are inspired by micheleg/dash-to-dock@8398d41
I saw they didn't look at the versions but at what is available, that's why I did that like that. I can change that of course, my main goal is to have this working on GNOME 3.30 :-)

Reply:
Thank you, I am not alone then :-)

I found this: `imports.misc.config.PACKAGE_VERSION`!

Reply:
That should be clearer now:

```diff
diff --git a/workspace-grid@mathematical.coffee.gmail.com/utils.js b/workspace-grid@mathematical.coffee.gmail.com/utils.js
index 8255046..1985931 100644
--- a/workspace-grid@mathematical.coffee.gmail.com/utils.js
+++ b/workspace-grid@mathematical.coffee.gmail.com/utils.js
@@ -18,14 +17,26 @@
  ***********************************************************************/
 
 const Meta = imports.gi.Meta;
+const ShellVersion = imports.misc.config.PACKAGE_VERSION.split(".");
+const ShellVersionId = ShellVersion[0] << 8 + ShellVersion[1];
+
+function isVersionAbove(major, minor) {
+    return ShellVersionId > (major << 8) + minor;
+}
 
 // Inspired by https://github.com/micheleg/dash-to-dock/commit/8398d41
 // Maintain compatibility with GNOME-Shell 3.30+ as well as previous versions.
 var WS = {
     getWS: function() {
-        return global.screen || global.workspace_manager;
+        if (isVersionAbove(3, 28)) {
+            return global.workspace_manager;
+        }
+        return global.screen;
     },
     getCorner: function() {
-        return Meta.ScreenCorner || Meta.DisplayCorner;
+        if (isVersionAbove(3, 28)) {
+            return Meta.DisplayCorner;
+        }
+        return Meta.ScreenCorner;
     }
 };
```